ui/src/components/VisualiseIcon/index.js

```diff
@@ -1,121 +1,138 @@
-      default: return '';
-    }
+  switch (type) {
+    case LEADERBOARD:
+    case TEMPLATE_MOST_ACTIVE_PEOPLE:
+    case TEMPLATE_MOST_POPULAR_ACTIVITIES:
+    case TEMPLATE_MOST_POPULAR_VERBS:
+      return LEADERBOARD_IMAGE;
+    case XVSY:
+      return XVSY_IMAGE;
+    case STATEMENTS:
+    case TEMPLATE_WEEKDAYS_ACTIVITY:
+      return STATEMENTS_IMAGE;
+    case FREQUENCY:
+    case TEMPLATE_ACTIVITY_OVER_TIME:
+      return FREQUENCY_IMAGE;
+    case COUNTER:
+    case TEMPLATE_LAST_7_DAYS_STATEMENTS:
+      return COUNTER_IMAGE;
+    case PIE:
+      return PIE_IMAGE;
+    default:
+      return '';
   }
+};
 
-  getIcon = (type, sourceView) => {
-    if (sourceView) {
-      return TABLE_IMAGE;
-    }
-    switch (type) {
-      case LEADERBOARD: return LEADERBOARD_IMAGE;
-      case XVSY: return XVSY_IMAGE;
-      case STATEMENTS: return STATEMENTS_IMAGE;
-      case FREQUENCY: return FREQUENCY_IMAGE;
-      case COUNTER: return COUNTER_IMAGE;
-      case PIE: return PIE_IMAGE;
-      case SESSIONS: return SESSIONS_IMAGE;
-      case PLATFORMS: return PLATFORMS_IMAGE;
-      case TABLE: return TABLE_IMAGE;
-      case QUESTIONANALYSIS: return QUESTIONANALYSIS_IMAGE;
-      default: return '';
-    }
+const VisualiseIcon = ({
+  type,
+  sourceView,
+  isSmall = true, // // [Viz Refactor] TODO: Remove this property
+}) => {
+  const src = getImageSrc(type, sourceView);
+  if (src === '') {
+    return null;
   }
 
-  renderIcon = ({ className } = {}) => {
-    const { type, sourceView } = this.props;
-    return (
-      <img
-        className={styles[className]}
-        src={this.getIcon(type, sourceView)}
-        alt={this.getTitle(type)} />
-    );
-  }
+  const classes = classNames({
+    [styles.visualisationSmall]: isSmall,
+  });
 
-  // return type !== 'undefined' ? (
-  //   <img
-  //     className={styles[className]}
-  //     src={this.getIcon(type)}
-  //     alt={this.getTitle(type)} />
-  // ) : <img
-  //   className={styles.visualisationSmall}
-  //   src={this.getIcon(SESSIONS)}
-  //   alt={this.getTitle(SESSIONS)} />;
+  return (
+    <img
+      className={classes}
+      src={src}
+      alt={getTitle(type)} />
+  );
+};
 
-  render = () => {
-    const { type, active, onClick, isSmall, className } = this.props;
-    const classes = classNames({
-      [styles['visualisation-icon']]: true,
-      [styles.active]: active
-    });
+VisualiseIcon.propTypes = {
+  sourceView: PropTypes.bool.isRequired,
+  type: PropTypes.string,
+  isSmall: PropTypes.bool,
+};
 
-    return (
-      isSmall
-      ? this.renderIcon({ className })
-      : (
-        <div className={classes} onClick={onClick} >
-          { this.renderIcon() }
-          <h5>{this.getTitle(type)}</h5>
-        </div>
-      )
-    );
-  }
-}
+export default withStyles(styles)(React.memo(VisualiseIcon));
+
+// [Viz Refactor] TODO: Remove VisualiseIconWithTitle after every types are refactored in TypeEditor
```

Comment:
I left some comments starting with `[Viz Refactor] TODO:`. These are tasks that should be done by the end of Viz Refactor project